azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachineScaleSet.java

```diff
@@ -0,0 +1,1042 @@
+     * @throws IOException exception thrown from serialization/deserialization
+     * @throws InterruptedException exception thrown when the operation is interrupted
+     */
+    void start() throws CloudException, IOException, InterruptedException;
+
+    /**
+     * Re-image (update the version of the installed operating system) the virtual machines in the scale set.
+     *
+     * @throws CloudException thrown for an invalid response from the service
+     * @throws IOException exception thrown from serialization/deserialization
+     * @throws InterruptedException exception thrown when the operation is interrupted
+     */
+    void reimage() throws CloudException, IOException, InterruptedException;
+
+    // Getters
+    //
+
+    /**
+     * @return the name prefix of the virtual machines in the scale set
+     */
+    String computerNamePrefix();
+
+    /**
+     * @return the operating system of the virtual machines in the scale set
+     */
+    OperatingSystemTypes osType();
+
+    /**
+     * @return the operating system disk caching type, valid values are 'None', 'ReadOnly', 'ReadWrite'
+     */
+    CachingTypes osDiskCachingType();
+
+    /**
+     * @return gets the name of the OS disk of virtual machines in the scale set
+     */
+    String osDiskName();
+
+    /**
+     * @return the upgradeModel
+     */
+    UpgradeMode upgradeModel();
+
+    /**
+     * @return true if over provision is enabled for the virtual machines, false otherwise
+     */
+    boolean overProvisionEnabled();
+
+    /**
+     * @return the sku of the virtual machines in the scale set
+     */
+    VirtualMachineScaleSetSkuTypes sku();
+
+    /**
+     * @return the number of virtual machine instances in the scale set
+     */
+    int capacity();
+
+    /**
+     * @return the virtual network associated with the primary network interfaces of the virtual machines
+     * in the scale set.
+     * <p>
+     * A primary internal load balancer associated with the primary network interfaces of the scale set
+     * virtual machine will be also belongs to this network
+     * </p>
+     * @throws IOException the IO exception
+     */
+    Network primaryNetwork() throws IOException;
+
+    /**
+     * @return the internet facing load balancer associated with the primary network interface of
+     * the virtual machines in the scale set.
+     *
+     * @throws IOException the IO exception
+     */
+    LoadBalancer primaryInternetFacingLoadBalancer() throws IOException;
+
+    /**
+     * @return the internet facing load balancer's backends associated with the primary network interface
+     * of the virtual machines in the scale set
+     *
+     * @throws IOException the IO exception
+     */
+    Map<String, Backend> primaryInternetFacingLoadBalancerBackEnds() throws IOException;
+
+    /**
+     * @return the internet facing load balancer's inbound NAT pool associated with the primary network interface
```

Comment:
"Internet-facing". 
I won't comment on these anymore -- it'd be good to just search/replace "internet facing" with "Internet-facing"